flashpay/apps/payments/tasks.py

```diff
@@ -0,0 +1,35 @@
+from base64 import b64decode
+from typing import Any
+
+from algosdk.error import IndexerHTTPError
+from huey import crontab
+from huey.contrib.djhuey import db_periodic_task, lock_task
+
+from django.conf import settings
+
+from flashpay.apps.payments.models import Transaction, TransactionStatus
+from flashpay.apps.payments.utils import verify_txn
+
+
+@db_periodic_task(crontab(minute="*/1"))
+@lock_task("lock-verify-txn")
+def verify_transactions() -> Any:
+    indexer = settings.INDEXER_CLIENT
+    transactions = Transaction.objects.filter(status=TransactionStatus.PENDING)
+    for transaction in transactions:
+        try:
+            results = indexer.search_transactions(
+                address=transaction.recipient, asset_id=transaction.asset.asa_id  # type: ignore
+            )
+            txns = results["transactions"]
```

Comment:
Why not just search for a single transaction?
Since we already know what to expect as a note, we can add that to the search query and use `.transaction()` instead, no?